### PR TITLE
Support app annotations and add one for Firefox Desktop

### PR DESCRIPTION
**Pull request by wlach**
* Indicate that the main reference for this app should still be the probe dictionary for now
* Provide a reference to Firefox on Glean

### CODE REVIEW

**Comment by wlach** [comment]
Tagging @chutten to review the specific annotation, @jklukas to review the updated ETL code.

File: annotations/firefox_desktop/README.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+---
+warning: >
+  Most Firefox Desktop telemetry is still being collected via the legacy telemetry
```
Comment (chutten) [inline annotations/firefox_desktop/README.md:3]:
```suggestion
  Most Firefox Desktop telemetry is still being collected via the legacy Firefox Telemetry
```

File: annotations/firefox_desktop/README.md [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+  collection system. A full list of this telemetry is available in the [probe dictionary](https://probes.telemetry.mozilla.org).
+---
+
+Most Firefox Desktop telemetry is still being collected via the legacy telemetry
```
Comment (chutten) [inline annotations/firefox_desktop/README.md:7]:
```suggestion
Most Firefox Desktop telemetry is still being collected via the legacy Firefox Telemetry
```

**Review by chutten — APPROVED** [review]
Basically branding nits around what to call things. It could land as-is without being too confusing, but we can be a little more precise.

The legacy system is Telemetry with a capital T. To make it distinct from "telemetry" (the noun referring to the kinds of data we most frequently collect) it's often referred to in a qualified form like "Firefox Telemetry" or "Firefox Desktop Telemetry".

File: scripts/create-api.py [hunk 3]
```diff
@@ -11,26 +11,38 @@
...
 apps = os.listdir(ANNOTATIONS_DIR)
+
+
+def read_annotation(filename):
+    annotation_md = frontmatter.load(filename)
+
+    annotation = {"content": annotation_md.content}
+    for key in ["component", "features", "warning"]:
+        if annotation_md.get(key):
+            annotation.update({key: annotation_md[key]})
```
Comment (jklukas) [inline scripts/create-api.py:22]:
```suggestion
            annotation[key] = annotation_md[key]
```

**Review by jklukas — APPROVED** [review]

**Pull request merged**